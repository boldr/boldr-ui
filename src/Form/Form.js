/* @flow */
import React from 'react';
import type { Node } from 'react';
import styled from 'styled-components';

export type Props = {
  children: Array<Node>,
  isInline: boolean,
  handleSubmit: () => void,
  className?: string,
};

export type MetaProps = {
  error?: string,
  warning?: string,
  touched?: boolean,
};

const FormElement = styled.form`display: ${props => (props.isInline ? 'inline-block' : 'block')};`;

const Form = (props: Props) => {
  const { handleSubmit, className } = props;

  return (
    <FormElement className={className} onSubmit={handleSubmit} {...props}>
      {props.children}
    </FormElement>
  );
};

Form.defaultProps = {
  isInline: false,
};

export default Form;
