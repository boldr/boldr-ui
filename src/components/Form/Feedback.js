// @flow
import React from 'react';
import styled from 'styled-components';

type Props = {
  meta: Object,
};

const Feedback = (props: Props) => {
  if (!props.meta) {
    return null;
  }

  const { meta } = props;
  const showError = meta.touched && meta.error && meta.invalid;
  return (
    <InputErrorContainer>

      {meta.error && <div className="boldrui-form__error">{meta.error}</div>}
      {meta.warning &&
        <div className="boldrui-form__warning">{meta.warning}</div>}
    </InputErrorContainer>
  );
};

export default Feedback;

const InputErrorContainer = styled.div`
  position: relative;
  top: -1.15rem;
`;

const InputErrorRed = styled.div`
  padding: 10px 20px;
  background: #fee7e8;
  color: #cc0726;
  font-size: 14px;
  border: 1px solid #cc0726;
  border-top: none;
  border-bottom-right-radius: 3px;
  border-bottom-left-radius: 3px;
  overflow: hidden;
`;
