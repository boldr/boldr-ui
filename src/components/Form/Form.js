/* @flow */
import React from 'react';
import styled from 'styled-components';
import type { ReactElement } from '../../types/react.js.flow';

type Props = {
  children: ReactElement,
  inline: boolean,
  handleSubmit: () => void,
};

export type MetaProps = {
  error: string,
  warning: string,
  touched: boolean,
};

const CustomForm = styled.form`
  display: {props => props.inline ? 'inline' : 'block'};
`;
const Form = (props: Props) => {
  const { handleSubmit } = props;

  return (
    <CustomForm className="boldrui-form" onSubmit={handleSubmit} {...props}>
      {props.children}
    </CustomForm>
  );
};

Form.defaultProps = {
  inline: false,
};

export default Form;
