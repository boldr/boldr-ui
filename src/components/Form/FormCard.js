/* @flow */
import React from 'react';
import Card from 'react-md/lib/Cards/Card';
import CardTitle from 'react-md/lib/Cards/CardTitle';
import CardActions from 'react-md/lib/Cards/CardActions';
import CardText from 'react-md/lib/Cards/CardText';
import styled from 'styled-components';

import type { ReactElement } from '../../types/react.js.flow';
import { Grid, Col, Row } from '../Layout';

const FormFooter = styled.ul`
  list-style-type: none;
`;

const FormFooterItem = styled.li`
  line-height: 1.5em;
`;

type Props = {
  width: Number,
  title: String,
  form: ReactElement,
  extra1: any,
  extra2: ?any,
};

const FormCard = (props: Props) => {
  return (
    <div>
      <Row>
        <Col xs={12}>
          <Row xsCenter>
            <Col xs={6}>
              <Card
                style={{
                  maxWidth: props.width || 600,
                  marginTop: '100px',
                }}
                className="md-block-centered"
              >
                <CardTitle className="boldr-form__title" title={props.title} expander />
                <CardText>
                  {props.form}
                </CardText>
                <CardActions style={{ justifyContent: 'center' }} expandable>
                  <FormFooter>
                    <FormFooterItem>{props.extra1}</FormFooterItem>
                    <FormFooterItem>{props.extra2}</FormFooterItem>
                  </FormFooter>
                </CardActions>
              </Card>
            </Col>
          </Row>
        </Col>
      </Row>
    </div>
  );
};

export default FormCard;
