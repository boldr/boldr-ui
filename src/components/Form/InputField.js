import React, { PropTypes } from 'react';
import TextField from 'react-md/lib/TextFields';

const propTypes = {
  input: PropTypes.object.isRequired,
  meta: PropTypes.object.isRequired,
  label: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  rows: PropTypes.number,
  maxRows: PropTypes.number,
};

const InputField = (
  {
    input,
    label,
    type,
    rows,
    maxRows,
    meta: { touched, error },
  },
) => {
  return (
    <TextField
      { ...input }
      id={ input.name }
      label={ label }
      placeholder={ input.placeholder }
      type={ type }
      errorText={ error }
      rows={ rows }
      maxRows={ maxRows }
    />
  );
};

InputField.propTypes = propTypes;

export default InputField;
