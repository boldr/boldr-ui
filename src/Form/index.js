import FormCard from './FormCard';
import FormGroup from './FormGroup';
import Form from './Form';
// import Label from './Label';
export { Form, FormCard, FormGroup };

export { default as Checkbox } from './Checkbox';
export { default as Control } from './Control';
export { default as Help } from './Help';
export { default as Input } from './Input';
export { default as Label } from './Label';
export { default as Radio } from './Radio';
export { default as SelectInput } from './SelectInput';
export { default as TextArea } from './TextArea';
// Form - Field
export { default as FormField } from './Field/FormField';
export { default as FieldBody } from './Field/FieldBody';
export { default as FieldLabel } from './Field/FieldLabel';
export {
  TextFormField,
  TextAreaFormField,
  SelectFormField,
  CheckboxFormField,
  RadioFormField,
} from './ReduxFormFields';

export { default } from './Form';
