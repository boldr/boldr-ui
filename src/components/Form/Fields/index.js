import InputField from './InputField';
import SelectInput from './SelectInput';
import RadioField from './RadioField';

export { InputField, RadioField, SelectInput };
