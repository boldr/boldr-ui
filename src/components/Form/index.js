import FormCard from './FormCard';
import FormGroup from './FormGroup';
import Form from './Form';
import { InputField, RadioField, SelectInput } from './Fields';
import Label from './Label';
import Feedback from './Feedback';
import InputWrapper from './InputWrapper';

export {
  Form,
  FormCard,
  FormGroup,
  InputField,
  RadioField,
  SelectInput,
  Label,
  Feedback,
  InputWrapper,
};
