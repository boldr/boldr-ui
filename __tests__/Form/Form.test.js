import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Form from '../../src/Form/Form';

const mockFn = jest.fn();

describe('<Form />', () => {
  const wrapper = shallow(<Form handleSubmit={mockFn} />);
  it('renders <Form /> without breaking', () => {
    expect(wrapper.find('.boldrui-form').length).toBe(1);
  });
  it('renders snapshot', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
