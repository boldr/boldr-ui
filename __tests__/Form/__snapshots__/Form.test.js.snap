// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> renders snapshot 1`] = `
<styled.form
  className="boldrui-form"
  handleSubmit={[Function]}
  inline={false}
  onSubmit={[Function]}
/>
`;
